app/jobs/submission_job.rb

```diff
@@ -33,7 +33,13 @@ def run_testcase(testcase)
 
   def calculate_result
     drives_candidate = @submission.drives_candidate
-    highest_submission = drives_candidate.submissions.order('total_marks desc').first
-    drives_candidate.update(score: highest_submission.total_marks)
+    problems = drives_candidate.drive.problems
+    score = 0
+    problems.each do |problem|
```

Comment:
Use collect and sum

Reply:
Here using collect is not straight forward as  each problem has multiple submissions, and we need one submissions with maximum marks

Reply:
```suggestion
    score = problems.collect do |problem|
      submissions = Submission.where(drives_candidate_id: drives_candidate.id, problem_id: problem.id)
      highest_submission = submissions.order('total_marks desc').first
      highest_submission.total_marks
    end.sum
```
won't this work ? @somnath77

Reply:
It worked, thanks